query_delete.go

```diff
@@ -286,3 +286,48 @@ func (q *DeleteQuery) afterDeleteHook(ctx context.Context) error {
 	}
 	return nil
 }
+
+//------------------------------------------------------------------------------
+type deleteQueryBuilder struct {
+	*DeleteQuery
+}
+
+func (q *deleteQueryBuilder) WhereGroup(sep string, fn func(QueryBuilder) QueryBuilder) QueryBuilder {
+	q.DeleteQuery.WhereGroup(sep, func(qs *DeleteQuery) *DeleteQuery {
```

Comment:
What about saving the return value `q.DeleteQuery = q.DeleteQuery` and everywhere else? In theory, the user can return a new `*DeleteQuery` from the callback.

Reply:
that makes sense, didn't realized the returned value, changing